src/components/Reports/GDPRPermissions/PermissionExpandableRow.js

```diff
@@ -0,0 +1,175 @@
+import React, { useState, useEffect } from 'react';
+import * as S from './SubscriberGdpr.styles';
+import { FormattedMessage, useIntl, FormattedDate } from 'react-intl';
+import PropTypes from 'prop-types';
+import { Loading } from '../../Loading/Loading';
+import { InjectAppServices } from '../../../services/pure-di';
+
+const PermissionValue = ({ value }) => {
+  const permissionValue = value.toLowerCase();
+  let iconColor =
+    permissionValue === 'none' ? 'grey' : permissionValue === 'true' ? 'green' : 'red';
+
+  return (
+    <div className="dp-icon-wrapper">
+      <span className={`ms-icon icon-lock dp-lock-${iconColor}`} />
+      <FormattedMessage id={`subscriber_gdpr.value_${permissionValue}`} />
+    </div>
+  );
+};
+
+const PermissionExpandableRow = ({
+  field,
+  email,
+  dependencies: {
+    dopplerApiClient,
+    appConfiguration: { reportsUrl },
+    experimentalFeatures,
+  },
+}) => {
+  const [expanded, setExpanded] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [permissions, setPermissions] = useState([]);
+  const intl = useIntl();
+  const _ = (id, values) => intl.formatMessage({ id }, values);
+  const isPermissionHistoryEnabled =
+    experimentalFeatures && experimentalFeatures.getFeature('PermissionHistory');
+
+  useEffect(() => {
+    const fetchData = async () => {
+      if (isPermissionHistoryEnabled) {
+        const fieldName = field.name;
+        const { success, value } = await dopplerApiClient.getSubscriberPermissionHistory({
+          email,
+          fieldName,
+        });
+        if (success) {
+          setPermissions(value.items);
+        }
+        setLoading(false);
```

Comment:
What happens when the result is not successful?

Reply:
@GusBaamonde what do you think? what should we do if there is an unexpected error?
By the moment @bravoy93, what do you think about adding a TODO comment?

```suggestion
        if (success) {
          setPermissions(value.items);
        }
        // TODO: deal with errors.
        setLoading(false);
```

Reply:
For the moment, it is out of the scope of the PR, so, I am closing this conversation...